src/panels/developer-tools/template/developer-tools-template.ts

```diff
@@ -281,6 +284,14 @@ class HaPanelDevTemplate extends LitElement {
             max(16px, env(safe-area-inset-left));
         }
 
+        .content.horizontal {
+          --code-mirror-max-height: calc(
+            100vh - var(--header-height) - var(--paper-font-body1_-_line-height) * 3 - 1em * 2
+            - max(16px,env(safe-area-inset-top)) * 2 - max(16px,env(safe-area-inset-bottom)) * 2
+            - var(--ha-card-border-width,1px) * 2 - 179px
+          );
```

Comment:
put this in `:host`, to be able to reuse it for the result panel.

When you use at the end a static px amount you can get rid of your unreadable variables and calculations you use. Because it still can lead to a needed manual change.

Since it is for desktop only, we also don't need `env(safe-area-inset-top)`

Reply:
> When you use at the end a static px amount you can get rid of your unreadable variables and calculations you use. Because it still can lead to a needed manual change.

According to what I see in Code Inspector, values for some properties are based on css variables (like `--header-height`, `--paper-font-body1_-_line-height`) - and other values are "hard-coded" (like "48px"). To make a layout "adaptable" to custom themes, I decided to use both css variables & exact values in calculations.

> Since it is for desktop only, we also don't need env(safe-area-inset-top)

This is for tablets also; so may be we should not get rid of these `env()`?

Reply:
> put this in :host, to be able to reuse it for the result panel.

What I just did in this [commit](https://github.com/home-assistant/frontend/commit/5e22cb2c7bf108a95ba92a07d82e73816de2f267) - simplified the 2nd occurrence of `calc()` - now it uses the `var(--code-mirror-max-height)` which is defined on a higher level.
I did not move a definition of `--code-mirror-max-height`, it is still defined in `.content.horizontal` (the minimal higher level in horiz layout).

Reply:
Sorry this is not maintainable. Just imagine you have to edit something in 6 months. Would you understand whats going on?

And as I said, it just makes sense when you use variables only, but when you put a static px amount at the end it isn't flexible anymore.